### PR TITLE
Fix wrong link to "Guarantees" part in introduction.html documentation

### PR DESCRIPTION
**Pull request by hojongs**
From Kafka 2.5 release, the id of html tag `Guarantees` part has changed from `#intro_guarantees` to `#impl_guarantees`
But, the links in introduction.html file are still linking to the `#intro_guarantees`
So, the link is not working. (do nothing) -> Needed fix.

I fixed link, but previous versions of the documentation (~ 2.4 releases) are still not working.
To fix the links of documentations of prior releases, we can change back the id from `#impl_guarantees` to `#intro_guarantees`. (not fix links, but change the id of the html tag) But I'm not sure Which way is better. please comment :)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/introduction.html [hunk 1]
```diff
@@ -144,13 +144,13 @@ <h4 class="anchor-heading">
     </li>
   </ul>
   <p>
-    <strong>Producers</strong> are those client applications that publish (write) events to Kafka, and <strong>consumers</strong> are those that subscribe to (read and process) these events. In Kafka, producers and consumers are fully decoupled and agnostic of each other, which is a key design element to achieve the high scalability that Kafka is known for. For example, producers never need to wait for consumers. Kafka provides various <a href="/documentation/#intro_guarantees">guarantees</a> such as the ability to process events exactly-once.
+    <strong>Producers</strong> are those client applications that publish (write) events to Kafka, and <strong>consumers</strong> are those that subscribe to (read and process) these events. In Kafka, producers and consumers are fully decoupled and agnostic of each other, which is a key design element to achieve the high scalability that Kafka is known for. For example, producers never need to wait for consumers. Kafka provides various <a href="/documentation/#impl_guarantees">guarantees</a> such as the ability to process events exactly-once.
```
Comment (mimaison) [inline docs/introduction.html]: I wonder if pointing to https://kafka.apache.org/documentation/#semantics would actually make more sense. WDYT?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR, I left a comment

File: docs/introduction.html (hunk 1 above)
Comment (hojongs) [reply docs/introduction.html]: You're right. That's more make sense.

Applied at f017014

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR

**Pull request merged**